### PR TITLE
Hotfix on pem key authentication. Also, Updated exception class.

### PR DESCRIPTION
**Pull request by esend7881**
- Swapped lines loading private key and connection appears to work now
- Updated exceptions to see more detailed stacktrace.
- Implementing integration test classes
- Bumped version to 2.1.1.3 to distinguish in nexus

### CODE REVIEW

File: src/main/java/net/juniper/netconf/Device.java [hunk 1]
```diff
@@ -246,12 +246,12 @@ private Session loginWithPrivateKey(int timeoutMilliSeconds) throws NetconfExcep
         try {
             Session session = sshClient.getSession(userName, hostName, port);
             session.setConfig("userauth", "publickey");
-            session.setConfig("StrictHostKeyChecking", "no");
+            session.setConfig("StrictHostKeyChecking", isStrictHostKeyChecking() ? "yes" : "no");
```
Comment (Vladislav-Kisliy) [inline src/main/java/net/juniper/netconf/Device.java:249]: it seems to me you forgot to add `isStrictHostKeyChecking()` method

**Review by Vladislav-Kisliy — CHANGES_REQUESTED** [review]
Sorry I didn't find any fixes and don't see reasons why we should add it

File: src/main/java/net/juniper/netconf/Device.java (hunk 1 above)
Comment (esend7881) [reply src/main/java/net/juniper/netconf/Device.java:249]: That method is auto-generated by lombok. It will compile

Comment (Vladislav-Kisliy) [reply src/main/java/net/juniper/netconf/Device.java:249]: oh, I see. I missed that, sorry. What about IT?

**Comment by esend7881** [comment]
I just added a function to run `<get>` netconf calls. I lumped it in here.

File: src/it/java/net/juniper/netconf/NetconfSessionIT.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@Category(IntegrationTest.class)
+public class NetconfSessionIT {
+
+    @Test //Expectation is no exception thrown.
+    public void testNetconfSessionWithKey() throws Exception {
+        try (Device device = Device.builder()
+                .hostName("localhost")
+                .userName(System.getProperty("user.name"))
+                .keyBasedAuthentication(true)
+                .pemKeyFile("/home/vagrant/.ssh/netconf_rsa")
```
[the remark this answers is not in the document]
Comment (peterjhill) [reply src/it/java/net/juniper/netconf/NetconfSessionIT.java:19]: yep.. for our own team and usage of this library, we have our own integration test, but it is not in the netconf library code.. it is in the code we wrap around the library.

I could see having something like this in the sample code directory in this repo. There may already be something like this. If you wanted to add more samples, that could be interesting.

File: src/main/java/net/juniper/netconf/Device.java [hunk 3]
```diff
@@ -273,7 +273,7 @@ private void loadPrivateKey() throws NetconfException {
         try {
             sshClient.addIdentity(pemKeyFile);
         } catch (JSchException e) {
-            throw new NetconfException(String.format("Error parsing the pemKeyFile: %s", e.getMessage()));
+            throw new NetconfException(String.format("Error parsing the pemKeyFile (%s): %s", getPemKeyFile(), e.getMessage()));
```
Comment (peterjhill) [inline src/main/java/net/juniper/netconf/Device.java:276]: the pem file is the private key, isn't it? I am not sure we should include the contents of the key file in the exception.. we don't want it showing up in a stack trace or log.

**Comment by esend7881** [comment]
I agree - I will update the error message and also remove my IT folder. I can keep that stuff on my local repo, no need to bother this one :)

**Comment by ydnath** [comment]
@esend7881 @peterjhill is this good for merging?

**Review by peterjhill — APPROVED** [review]

**Pull request merged**